### ISSUE TITLE
WaveRNN interface

### ISSUE THREAD

**Issue by m-toman** [Rayhane-mamah/Tacotron-2#180]
Hi,

I've trained the WaveRNN implementation from this repo: https://github.com/fatchord/WaveRNN with GTA mel specs from your repo and it worked pretty well for LJ:
https://www.dropbox.com/sh/2gtunx8d1r92fqb/AADh9CJEtvHnQ7YlwNClk8X5a?dl=0
(those are only GTA samples, didn't run it fully end-to-end yet)

My repository including pretrained models is here: https://github.com/m-toman/tacorn
It's not implemented nicely yet, just did what's necessary to try it out.

Is there a reason you are using 24kHz in this version?

**Comment by PetrochukM** [comment]
This is not WaveRNN, WaveRNN has 16-bit fidelity while fatchord's model has 9-bit fidelity. This is fatchord's heavily modified WaveRNN that does not sound quite as good as WaveNet. Fatchord said that himself.

**Comment by m-toman** [comment]
I should probably mention that it's his modified version, yes.
But I'm pretty happy with the 718k steps version in https://www.dropbox.com/sh/2gtunx8d1r92fqb/AADh9CJEtvHnQ7YlwNClk8X5a?dl=0

I only briefly tried the nvidia implementation and encountered a number of issues I'd have to solve first. But my main issue with it is that it seems very difficult to deploy on anything else than specific nvidia GPUs. 
With WaveRNN I don't think it's easy to reproduce the original optimizations in the paper, but at least there is a slight chance to run it on consumer hardware, be it a browser or an app.

Not sure what about http://gfx.cs.princeton.edu/pubs/Jin_2018_FAR/

**Comment by PetrochukM** [comment]
It's not really reduced training time. R9Y9 trains a 16-bit WaveNet implementation that is higher fidelity. That is not an apple to apple comparison. Try R9Y9's 8-bit settings, they should train similarly fast. 

WaveNet can also be run on a CPU, see...
http://on-demand.gputechconf.com/gtc/2017/presentation/s7544-andrew-gibiansky-efficient-inference-for-wavenet.pdf

With 95% sparsification added to WaveNet, and the above kernel, it can also be run on a mobile CPU with real time.

**Comment by m-toman** [comment]
Thanks for the slides, I don't have time at the moment to check them out but on a first glance look very helpful. 

Regarding training time I was just citing fatchord
"The first is WaveRNN, however it is quite slow to train (~7 days).

The good news is that I came up with another model that trains much faster and can handle the noise in predicted features from Tacotron and similar models".

I mainly wanted to try it out because it was so easy to do (probably half an hour to feed the GTAs to the jupyter notebook.
But I see that most new development happens with Wavenet. I didn't find much more time than to read the original paper and also the WaveRNN paper. I only skimmed the DeepVoice, FFTNet, Clarinet etc papers.
